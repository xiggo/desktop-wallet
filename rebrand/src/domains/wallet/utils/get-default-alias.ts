import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let counter = sameCoinWallets.length;

	if (counter === 0) {
		counter = 1;
	}

	while (profile.wallets().findByAlias(makeAlias(counter))) {
		counter++;
	}

	return makeAlias(counter);
};
