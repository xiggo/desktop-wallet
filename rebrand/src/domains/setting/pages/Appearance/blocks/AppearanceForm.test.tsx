import React from "react";

import { AppearanceForm } from "./AppearanceForm";
import { env, render, screen } from "@/utils/testing-library";

describe("AppearanceForm", () => {
	it("should render", () => {
		const profile = env.profiles().create("empty profile");

		const { asFragment } = render(<AppearanceForm profile={profile} />);

		expect(screen.getAllByRole("radiogroup")).toHaveLength(2);

		expect(screen.getByTestId("AppearanceFooterButtons__save")).toBeDisabled();

		expect(asFragment()).toMatchSnapshot();
	});
});
