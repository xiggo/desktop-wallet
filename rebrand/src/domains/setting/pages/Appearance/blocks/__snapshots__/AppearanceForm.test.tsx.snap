// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppearanceForm should render 1`] = `
<DocumentFragment>
  <form
    class="mt-8"
    data-testid="AppearanceForm"
  >
    <ul
      class="sc-hKwDye kOdhou"
      data-testid="list-divided__items"
    >
      <li
        class="flex flex-col w-full pb-6"
        data-testid="list-divided-item__wrapper"
      >
        <div
          class="flex justify-between items-center"
          data-testid="list-divided-item__inner-wrapper"
        >
          <div
            class="flex flex-col space-y-2 w-full"
          >
            <div
              class="flex justify-between items-center space-x-5"
            >
              <span
                class="font-semibold text-theme-secondary-700 dark:text-theme-secondary-200"
                data-testid="list-divided-item__label"
              >
                Accent Color
              </span>
            </div>
            <span
              class="text-sm font-medium text-theme-secondary-500 dark:text-theme-secondary-700"
              data-testid="list-divided-item__label--description"
            >
              Choose a color for buttons and other wallet features.
            </span>
          </div>
          <div>
            <span
              data-testid="list-divided-item__value"
            >
              <div
                class="inline-flex items-center space-x-3 w-full"
                data-testid="ButtonGroup"
                role="radiogroup"
              >
                <button
                  aria-checked="false"
                  aria-label="Blue"
                  class="ButtonGroup__ButtonGroupOptionStyled-sc-2f0qmd-0 lfZjmj"
                  data-testid="ButtonGroupOption"
                  role="radio"
                  type="button"
                >
                  <div
                    class="w-5 h-5 mx-1 rounded bg-theme-info-600"
                  />
                </button>
                <button
                  aria-checked="true"
                  aria-label="Green"
                  class="ButtonGroup__ButtonGroupOptionStyled-sc-2f0qmd-0 lfZjmj"
                  data-testid="ButtonGroupOption"
                  role="radio"
                  type="button"
                >
                  <div
                    class="w-5 h-5 mx-1 rounded bg-theme-success-600"
                  />
                </button>
              </div>
            </span>
          </div>
        </div>
      </li>
      <li
        class="flex flex-col w-full py-6"
        data-testid="list-divided-item__wrapper"
      >
        <div
          class="flex justify-between items-center"
          data-testid="list-divided-item__inner-wrapper"
        >
          <div
            class="flex flex-col space-y-2 w-full"
          >
            <div
              class="flex justify-between items-center space-x-5"
            >
              <span
                class="font-semibold text-theme-secondary-700 dark:text-theme-secondary-200"
                data-testid="list-divided-item__label"
              >
                Viewing Mode
              </span>
            </div>
            <span
              class="text-sm font-medium text-theme-secondary-500 dark:text-theme-secondary-700"
              data-testid="list-divided-item__label--description"
            >
              Choose a light or dark viewing mode.
            </span>
          </div>
          <div>
            <span
              data-testid="list-divided-item__value"
            >
              <div
                class="inline-flex items-center space-x-3 w-full"
                data-testid="ButtonGroup"
                role="radiogroup"
              >
                <button
                  aria-checked="true"
                  aria-label="light"
                  class="ButtonGroup__ButtonGroupOptionStyled-sc-2f0qmd-0 lfZjmj"
                  data-testid="ButtonGroupOption"
                  role="radio"
                  type="button"
                >
                  <div
                    class="flex items-center space-x-2 px-2"
                  >
                    <div
                      class="sc-bdvvtL qGXhc dark:text-theme-secondary-600"
                      height="20"
                      width="20"
                    >
                      <svg>
                        underline-sun.svg
                      </svg>
                    </div>
                    <span>
                      Light
                    </span>
                  </div>
                </button>
                <button
                  aria-checked="false"
                  aria-label="dark"
                  class="ButtonGroup__ButtonGroupOptionStyled-sc-2f0qmd-0 lfZjmj"
                  data-testid="ButtonGroupOption"
                  role="radio"
                  type="button"
                >
                  <div
                    class="flex items-center space-x-2 px-2"
                  >
                    <div
                      class="sc-bdvvtL qGXhc dark:text-theme-secondary-600"
                      height="20"
                      width="20"
                    >
                      <svg>
                        underline-moon.svg
                      </svg>
                    </div>
                    <span>
                      Dark
                    </span>
                  </div>
                </button>
              </div>
            </span>
          </div>
        </div>
      </li>
      <li
        class="flex flex-col w-full py-6"
        data-testid="list-divided-item__wrapper"
      >
        <div
          class="flex justify-between items-center"
          data-testid="list-divided-item__inner-wrapper"
        >
          <div
            class="flex flex-col space-y-2 w-full"
          >
            <div
              class="flex justify-between items-center space-x-5"
            >
              <span
                class="font-semibold text-theme-secondary-700 dark:text-theme-secondary-200"
                data-testid="list-divided-item__label"
              >
                Portfolio Latest Transactions
              </span>
              <span>
                <label
                  class="Toggle__Wrapper-sc-6s3wfu-0 efNWJo"
                >
                  <input
                    class="Toggle__Input-sc-6s3wfu-1 WwplI"
                    data-testid="AppearanceToggle__toggle-dashboardTransactionHistory"
                    name="dashboardTransactionHistory"
                    type="checkbox"
                  />
                  <div
                    class="Toggle__Handle-sc-6s3wfu-2 cobHFC"
                  >
                    <span
                      class="Toggle__HandleInner-sc-6s3wfu-3 ddepvC"
                    />
                  </div>
                </label>
              </span>
            </div>
            <span
              class="text-sm font-medium text-theme-secondary-500 dark:text-theme-secondary-700"
              data-testid="list-divided-item__label--description"
            >
              Enables latest transactions on Portfolio page.
            </span>
          </div>
        </div>
      </li>
      <li
        class="flex flex-col w-full py-6"
        data-testid="list-divided-item__wrapper"
      >
        <div
          class="flex justify-between items-center"
          data-testid="list-divided-item__inner-wrapper"
        >
          <div
            class="flex flex-col space-y-2 w-full"
          >
            <div
              class="flex justify-between items-center space-x-5"
            >
              <span
                class="font-semibold text-theme-secondary-700 dark:text-theme-secondary-200"
                data-testid="list-divided-item__label"
              >
                Wallet Naming
              </span>
              <span>
                <label
                  class="Toggle__Wrapper-sc-6s3wfu-0 efNWJo"
                >
                  <input
                    class="Toggle__Input-sc-6s3wfu-1 WwplI"
                    data-testid="AppearanceToggle__toggle-useNetworkWalletNames"
                    name="useNetworkWalletNames"
                    type="checkbox"
                  />
                  <div
                    class="Toggle__Handle-sc-6s3wfu-2 cobHFC"
                  >
                    <span
                      class="Toggle__HandleInner-sc-6s3wfu-3 ddepvC"
                    />
                  </div>
                </label>
              </span>
            </div>
            <span
              class="text-sm font-medium text-theme-secondary-500 dark:text-theme-secondary-700"
              data-testid="list-divided-item__label--description"
            >
              Prioritize showing network provided names over wallet names.
            </span>
          </div>
        </div>
      </li>
      <li
        class="flex flex-col w-full pt-6"
        data-testid="list-divided-item__wrapper"
      >
        <div
          class="flex justify-between items-center"
          data-testid="list-divided-item__inner-wrapper"
        >
          <div
            class="flex flex-col space-y-2 w-full"
          >
            <div
              class="flex justify-between items-center space-x-5"
            >
              <span
                class="font-semibold text-theme-secondary-700 dark:text-theme-secondary-200"
                data-testid="list-divided-item__label"
              >
                Expanded Tables
              </span>
              <span>
                <label
                  class="Toggle__Wrapper-sc-6s3wfu-0 efNWJo"
                >
                  <input
                    class="Toggle__Input-sc-6s3wfu-1 WwplI"
                    data-testid="AppearanceToggle__toggle-useExpandedTables"
                    name="useExpandedTables"
                    type="checkbox"
                  />
                  <div
                    class="Toggle__Handle-sc-6s3wfu-2 cobHFC"
                  >
                    <span
                      class="Toggle__HandleInner-sc-6s3wfu-3 ddepvC"
                    />
                  </div>
                </label>
              </span>
            </div>
            <span
              class="text-sm font-medium text-theme-secondary-500 dark:text-theme-secondary-700"
              data-testid="list-divided-item__label--description"
            >
              Increases spacing of table items.
            </span>
          </div>
        </div>
      </li>
    </ul>
    <div
      class="flex justify-end mt-8 space-x-3 w-full"
    >
      <button
        class="OriginalButton__StyledButton-sc-o0fs2j-0 kkDhXe"
        data-testid="AppearanceFooterButtons__save"
        disabled=""
        type="submit"
      >
        <div
          class="flex items-center space-x-2"
        >
          Save
        </div>
      </button>
    </div>
  </form>
</DocumentFragment>
`;
